### PR TITLE
feat(kinput): add characterLimit

### PR DESCRIPTION
**Pull request by adamdehaven**
### Summary

Add `characterLimit` prop to `KInput`

### PR Checklist
- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: docs/components/input.md [hunk 1]
```diff
@@ -14,10 +14,10 @@
 
 To set the value of the input element without using `v-model`, you can set the `model-value` attribute on the input:
 
-<KInput model-value="This is the input value" />
+<KInput model-value="This is the input value" character-limit="10" placeholder="Placeholder text" />
```
Comment (kaiarrowood) [inline docs/components/input.md:17]: I think you want to revert this change

File: docs/components/input.md [hunk 2]
```diff
@@ -100,6 +100,29 @@ You also have the option of using the `.help` utility class. This is meant to be
...
+
+Use this prop to specify a character limit for the input.
+
+<KInput model-value="This field has too many characters" :character-limit="10" class="w-100" placeholder="Placeholder text" />
+
+```vue
+<KInput model-value="This field has too many characters" :character-limit="10" class="w-100" placeholder="Placeholder text" />
+```
+
+If the `characterLimit` is exceeded, the character counter below the `KInput` will override the display of a provided `errorMessage` until the character count is within the acceptable range.
```
Comment (kaiarrowood) [inline docs/components/input.md:113]: May want to note that the character counter is only visible when the char limit is exceeded

File: docs/components/input.md [hunk 3]
```diff
@@ -108,13 +131,13 @@ Boolean value to indicate whether the element has an error and should apply erro
 
 String to be displayed as error message if `hasError` prop is `true`.
 
-<KInput class="w-100" hasError errorMessage="Service name should not contain '_'"/>
+<KInput class="w-100" hasError errorMessage="Service name should not contain '_'" character-limit="10"/>
```
Comment (kaiarrowood) [inline docs/components/input.md:134]: I think you want to revert this change?

File: src/components/KInput/KInput.vue [hunk 4]
```diff
@@ -19,22 +19,22 @@
         <input
           v-bind="modifiedAttrs"
           :id="inputId"
-          :value="currValue ? currValue : modelValue"
+          :value="currValue || modelValueChanged ? currValue : modelValue"
           :class="`k-input-${size}`"
-          :aria-invalid="hasError ? hasError : 'false'"
+          :aria-invalid="hasError ? hasError : charLimitExceeded ? 'true' : 'false'"
```
Comment (kaiarrowood) [inline src/components/KInput/KInput.vue:24]:
```suggestion
          :aria-invalid="hasError || charLimitExceeded ? 'true' : 'false'"
```

File: src/components/KTextArea/KTextArea.vue [hunk 5]
```diff
@@ -31,7 +31,7 @@
           :value="currValue ? currValue : modelValue"
           :rows="rows"
           :cols="cols"
-          :aria-invalid="hasError ? hasError : 'false'"
+          :aria-invalid="hasError ? hasError : charLimitExceeded ? 'true' : 'false'"
```
Comment (kaiarrowood) [inline src/components/KTextArea/KTextArea.vue:34]:
```suggestion
          :aria-invalid="hasError || charLimitExceeded ? 'true' : 'false'"
```

**Review by kaiarrowood — CHANGES_REQUESTED** [review]

File: docs/components/input.md (hunk 2 above)
Comment (adamdehaven) [reply docs/components/input.md:115]: Added

File: docs/components/input.md (hunk 1 above)
Comment (adamdehaven) [reply docs/components/input.md]: Done

File: docs/components/input.md (hunk 3 above)
Comment (adamdehaven) [reply docs/components/input.md]: Done

File: src/components/KTextArea/KTextArea.vue (hunk 5 above)
Comment (adamdehaven) [reply src/components/KTextArea/KTextArea.vue]: fixed, see comment above

File: src/components/KTextArea/KTextArea.vue [hunk 6]
```diff
@@ -58,7 +58,7 @@
         :value="currValue ? currValue : modelValue"
         :rows="rows"
         :cols="cols"
-        :aria-invalid="hasError ? hasError : 'false'"
+        :aria-invalid="hasError ? hasError : charLimitExceeded ? 'true' : 'false'"
```
[the remark this answers is not in the document]
Comment (adamdehaven) [reply src/components/KTextArea/KTextArea.vue]: fixed, see comment above

File: docs/components/input.md [hunk 7]
```diff
@@ -2,10 +2,10 @@
 
 **KInput** provides a wrapper around general `text` input's and provides specific Kong styling and state treatments (error, focus, etc).
 
-<KInput class="w-100" placeholder="Placeholder text..."/>
+<KInput class="w-100" placeholder="Placeholder text" :maxlength="10" character-limit="10" label="Test label" help="This is help text"/>
```
[the remark this answers is not in the document]
Comment (adamdehaven) [reply docs/components/input.md]: Done

**Review by kaiarrowood — APPROVED** [review]
Looks good! 🚀
